### ISSUE TITLE
Setting the serial tx buffer size to 255 "breaks" serial send

### ISSUE THREAD

**Issue by microbit-carlos** [lancaster-university/codal-microbit-v2#74]
Because the codal-core driver adds one, overflowing the uint8_t:

```cpp
int Serial::setTxBufferSize(uint8_t size)
{
    if(txInUse())
        return DEVICE_SERIAL_IN_USE;

    lockTx();

    // + 1 so there is a usable buffer size, of the size the user requested.
    this->txBuffSize = size + 1;

    int result = initialiseTx();

    unlockTx();

    return result;
}
```

**Comment by finneyj** [comment]
Thanks @microbit-carlos 

Fixed in https://github.com/lancaster-university/codal-core/commit/8c0e1e475b97dedbdb43d76a015220ff091cfba9.

**Comment by finneyj** [comment]
Fixed in v0.2.25
